test/test_adb_device.py

```diff
@@ -0,0 +1,78 @@
+import os.path
+import shutil
+import unittest
+from os.path import basename
+from pathlib import Path
+from subprocess import CompletedProcess
+from unittest.mock import patch, Mock
+
+from parameterized import parameterized
+
+from adb_pywrapper.adb_device import AdbDevice
+from adb_pywrapper.adb_result import AdbResult
+
+
+class MockAdbResult(AdbResult):
+    def __init__(self, stdout: str = '', stderr: str = '', success: bool = True):
+        self.completed_adb_process = Mock(CompletedProcess)
+        self.stdout = stdout
+        self.stderr = stderr
+        self.success = success
+
+
+class TestAdbDevice(unittest.TestCase):
+
+    @parameterized.expand([
+        ('/sdcard/test.txt', './foo', 'pull /sdcard/test.txt ./foo'),
```

Comment:
Could you make is more clear what the testcases are? Maybe add a comment or work the testcase into the case. E.g. 'this/is/a/case/with /a/space'

Reply:
Maybe add a reference to the testcase for this bug?